File: challenges/3-tipos-de-triangulo/ruby/jailsonsf/desafio-3.rb
inputs = ARGV.map(&:to_f)
sorted = inputs.sort { |a, b| b <=> a }

A = sorted[0]
B = sorted[1]
C = sorted[2]

puts "Entrada: #{inputs[0]} #{inputs[1]} #{inputs[2]}"

if A >= B + C
  puts 'NAO FORMA TRIANGULO'
else
  puts 'TRIANGULO RETANGULO' if A == B + C
  puts 'TRIANGULO OBTUSANGULO' if A > B + C
  puts 'TRIANGULO ACUTANGULO' if A < B + C
  puts 'TRIANGULO EQUILATERO' if A == B && B == C
  puts 'TRIANGULO ISOSCELES' if (A == B && B != C) || (A == C && C != B) || (C == B && B != A)
end

File: challenges/3-tipos-de-triangulo/python/I-Am-Not-Lucas/desafio_3.py
lista_lados = []

a = float(input("Lado a >>"))
b = float(input("Lado b >>"))
c = float(input("Lado c >>"))


lista_lados.append(a)
lista_lados.append(b)
lista_lados.append(c)
lista_lados.sort(reverse=True)

print("Entrada: ", end='')
for valor in lista_lados:
    print(valor, end=' ')
print()
print()

def nomenclatura1(lista):
    if lista_lados[0] >= lista_lados[1] + lista_lados[2]:
        print("NAO FORMA TRIANGULO")

    elif lista_lados[0] ** 2 == lista_lados[1] **2 + lista_lados[2] **2:
        print("TRIANGULO RETANGULO")

    elif lista_lados[0] ** 2 > lista_lados[1] **2 + lista_lados[2] **2:
        print("TRIANGULO OBTUSANGULO")

    elif lista_lados[0] ** 2 < lista_lados[1] **2 + lista_lados[2] **2:
        print("TRIANGULO ACUTANGULO")

def nomenclatura2 (n1, n2, n3):
    equilatero = False

    if n1 == n2 and n2 == n3:
        print("TRIANGULO EQUILATERO")
        equilatero = True

    elif n1 == n2 or n2 == c or c == n1 and not equilatero:
        print("TRIANGULO ISOSCELES")

nomenclatura1(lista_lados)
nomenclatura2(a, b, c)


